src/main/java/ch/njol/skript/events/EvtEntityPotion.java

```diff
@@ -69,14 +70,15 @@ public boolean check(Event event) {
 		PotionEffectType effectType = potionEvent.getNewEffect().getType();
 		for (PotionEffectType potionEffectType : potionEffects.getArray(event)) {
 			if (potionEffectType.equals(effectType)) {
-				return true;
+				if (cause == null || cause.getSingle(event).equals(potionEvent.getCause())) {
+					return true;
+				}
 			}
 		}
```

Comment:
What if there's a cause but no potion effect specified?

Reply:
Also have you tested without causes/effects? I think you need to make the patterns nullable (%-type%)